src/main/java/org/openrewrite/java/testing/junit6/MinimumJreConditions.java

```diff
@@ -0,0 +1,412 @@
+                    if (disabledOnJre.get().stream().allMatch(v -> compareVersions(v, javaVersion) < 0)) {
+                        m = removeAnnotation(m, DISABLED_ON_JRE, ctx);
+                    } else {
+                        m = updateAnnotationVersions(m, disabledOnJre, DISABLED_ON_JRE, ctx);
+                    }
+                }
+                if (enabledOnJreRange.isPresent()) {
+                    Range range = enabledOnJreRange.get();
+                    if (compareVersions(range.getMax(), javaVersion) < 0) {
+                        // Remove the test method if it is enabled on a JRE range that ends before the specified version
+                        maybeRemoveImport(JRE_IMPORT);
+                        return null;
+                    }
+                    if (compareVersions(range.getMin(), javaVersion) <= 0 && range.getMaxNotation() == null) {
+                        // The test is enabled for all versions from `javaVersion till end -> remove annotation
+                        m = removeAnnotation(m, ENABLED_FOR_JRE_RANGE, ctx);
+                    } else {
+                        m = updateRangeStart(m, ENABLED_FOR_JRE_RANGE, range, ctx);
+                    }
+                }
+                if (disabledOnJreRange.isPresent()) {
+                    Range range = disabledOnJreRange.get();
+                    if (compareVersions(range.getMax(), javaVersion) < 0) {
+                        m = removeAnnotation(m, DISABLED_FOR_JRE_RANGE, ctx);
+                    } else if (compareVersions(range.getMin(), javaVersion) <= 0 && range.getMaxNotation() == null) {
+                        // The test is disabled for all versions from `javaVersion` till end -> remove test
+                        maybeRemoveImport(JRE_IMPORT);
+                        return null;
+                    } else {
+                        m = updateRangeStart(m, DISABLED_FOR_JRE_RANGE, range, ctx);
+                    }
+                }
+
+                replaceSingleVersionRangeWithEquivalentAnnotation(enabledOnJreRange, ENABLED_FOR_JRE_RANGE, ENABLED_ON_JRE);
+                replaceSingleVersionRangeWithEquivalentAnnotation(disabledOnJreRange, DISABLED_FOR_JRE_RANGE, DISABLED_ON_JRE);
+
+                if (m != method) {
+                    return simplifySingleValueAnnotationAttributeArrays(m, prefix);
+                }
+                return method;
+            }
+
+            private J.MethodDeclaration removeAnnotation(J.MethodDeclaration m, String annotationType, ExecutionContext ctx) {
+                RemoveAnnotation removeAnnotation = new RemoveAnnotation("@" + annotationType);
+                maybeRemoveImport(JRE_IMPORT);
+                return removeAnnotation.getVisitor().visitMethodDeclaration(m, ctx);
+            }
+
+            private J.MethodDeclaration updateAnnotationVersions(J.MethodDeclaration m, Optional<List<String>> versions, String annotationType, ExecutionContext ctx) {
```

Comment:
```suggestion
            private J.MethodDeclaration updateAnnotationVersions(J.MethodDeclaration m, List<String> versions, String annotationType, ExecutionContext ctx) {
```

Don't use Optional as input argument of a function. In this case it's unnecessary, as you are sure the optional is always true (thus handle such case on the calling side).

On a more general level, it is seen as bad practice to use the Optional as method parameter, as it was never intended as a Maybe type. See for more info https://stackoverflow.com/questions/26327957/should-java-8-getters-return-optional-type/26328555#26328555 (also note here that Brian Goetz is not 'just' another programmer 😉).

--

In addition to this: I think, now we have JSpecify, we can use the Optional type to an even lesser extent. From a functional side, `@Nullable Object` or `Optional<Object>` concern themselves with the same idea, namely marking a type as `T | null`.